frappe/patches/v14_0/save_ratings_in_fraction.py

```diff
@@ -1,12 +1,52 @@
 import frappe
+from frappe.query_builder import DocType
+
 
 def execute():
-	rating_fields = frappe.get_all("DocField", fields=["parent", "fieldname"], filters={"fieldtype": "Rating"})
+	RATING_FIELD_TYPE = "decimal(3,2)"
+	rating_fields = frappe.get_all(
+		"DocField", fields=["parent", "fieldname"], filters={"fieldtype": "Rating"}
+	)
+
+	custom_rating_fields = frappe.get_all(
+		"Custom Field", fields=["dt", "fieldname"], filters={"fieldtype": "Rating"}
+	)
+
+	for _field in rating_fields + custom_rating_fields:
+		doctype_name = _field.get("parent") or _field.get("dt")
+		doctype = DocType(doctype_name)
+		field = _field.fieldname
+
+		# TODO: Add postgres support (for the check)
+		if (
+			frappe.conf.db_type == "mariadb"
+			and frappe.db.get_column_type(doctype_name, field) == RATING_FIELD_TYPE
+		):
+			continue
+
+		# update NULL values to 0 to avoid data truncated error (temp)
+		# commit for upcoming DLL
+		frappe.qb.update(doctype).set(
```

Comment:
This is not required. I think the 'data truncated' error is from hard coded `NOT NULL` in the `db.change_column_type` 🤦 If we permit null (like we should in this case) then conversion goes through without any problem.

Reply:
IMO this method should be refactored to add 2 parameters: nullable and default which are frequently required when altering column.